src/com/can/CANHandler.cpp

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Alexander Trojnin github.com/trojninalex
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Alexander Trojnin - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+
+#include <sockhand.h>
+
+#include "fdselecthand.h"
+#include "devlog.h"
+#include "devexec.h"
+#include "commfb.h"
+#include "comCallback.h"
+#include "criticalregion.h"
+
+#include "CANHandler.h"
+
+DEFINE_HANDLER(CCANHandler);
+
+CCANHandler::CCANHandler(CDeviceExecution &paDeviceExecution) : CExternalEventHandler(paDeviceExecution)
+{
+    mConnectionListChanged = false;
+}
+
+CCANHandler::~CCANHandler()
+{
+    mConnectionsList.clear();
+
+    this->end();
+}
+
+void CCANHandler::addComCallback(CFDSelectHandler::TFileDescriptor paFD, forte::com_infra::CComCallback *paCanLayer)
+{
+    DEVLOG_DEBUG("[CAN Handler] add callback");
```

Comment:
Newline missing. Maybe improve the message a bit like "Callback handler added"

Reply:
Update